new-new-packages/compiler/src/compiler/string_to_rcst.rs

```
@@ -0,0 +1,1394 @@
+                    _ => panic!(
+                        "single_line_whitespace returned something other than Whitespace or Error."
+                    ),
+                };
+                expressions.push(Rcst::Error {
+                    unparsable_input: unexpected_whitespace.to_string(),
+                    error: RcstError::TooMuchWhitespace,
+                });
+            }
+
+            match expression(input, indentation, true) {
+                Some((new_input, expression)) => {
+                    input = new_input;
+                    expressions.push(expression);
+                }
+                None => {
+                    let fallback = colon(new_input)
+                        .or_else(|| comma(new_input))
+                        .or_else(|| closing_parenthesis(new_input))
+                        .or_else(|| closing_bracket(new_input))
+                        .or_else(|| closing_curly_brace(new_input))
+                        .or_else(|| arrow(new_input));
+                    if let Some((new_input, cst)) = fallback {
+                        input = new_input;
+                        expressions.push(cst);
+                    }
+                }
+            }
+        }
+        (input, expressions)
+    }
+
+    fn lambda(input: &str, indentation: usize) -> Option<(&str, Rcst)> {
+        log::trace!("lambda({:?}, {:?})", input, indentation);
+        let (input, mut opening_curly_brace) = opening_curly_brace(input)?;
+        let (mut input, mut parameters_and_arrow) = {
+            let input_without_params = input;
+            let mut input = input;
+            let mut parameters = vec![];
+            loop {
+                let (i, whitespace) = whitespaces_and_newlines(input, indentation + 1, true);
+                if parameters.is_empty() {
+                    opening_curly_brace = opening_curly_brace.wrap_in_whitespace(whitespace);
+                }
```

If there are already some parameters, the whitespace is silently discarded